### PR TITLE
1375 fix stray tap file bricking install

### LINKED ISSUE #1375: Problem during installation
**Describe the bug**
I tried to install a package without specifying the `--version` it failed, I added the `--version` still failed.

**Environment (please complete the following information):**
 - OS: Windows 11
 - OpenTAP Version 9.22.1+9447a2af

**To Reproduce**
Not sure how to reproduce it, but I kept the installation folder for inspection.

**Actual behavior**
```bash
tap package install Runner 1.7.1

Unhandled Exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.ComponentModel.Win32Exception: The specified executable is not a valid application for this OS platform.
   at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at OpenTap.ExecutorSubProcess.Start()
   at OpenTap.Cli.TapEntry.goIsolated()
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at tap.Program.Main(String[] args)
^C
```

```
C:\TEMP\RunnerLocal171>tap package install Runner --version 1.7.1

Unhandled Exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.ComponentModel.Win32Exception: The specified executable is not a valid application for this OS platform.
   at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at OpenTap.ExecutorSubProcess.Start()
   at OpenTap.Cli.TapEntry.goIsolated()
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at tap.Program.Main(String[] args)
```

**Expected behavior**
The installation happens.

**Additional context**
[SessionLogs.zip](https://github.com/opentap/opentap/files/13869975/SessionLogs.zip)

### PR DESCRIPTION
**Pull request by alnlarsen**
This fixes an issue which causes `tap package install` to break on Windows if a file named `tap` is placed in the installation directory.

Closes #1375

### CODE REVIEW

File: Engine/Annotations/Annotation.cs [hunk 1]
```diff
@@ -1810,18 +1791,23 @@ public void Write(object source)
                         rdonly = true;
                     if (!rdonly)
                     {
-                        lst2.Clear();
-
+                        // Arrays must be re-allocated
```
Comment (rmadsen-ks) [inline Engine/Annotations/Annotation.cs:1794]: Should these changes be in a separate PR?

**Review by rmadsen-ks — CHANGES_REQUESTED** [review]
It seems there is one too many commits in this PR. otherwise looks good.

File: Engine/Annotations/Annotation.cs (hunk 1 above)
Comment (alnlarsen) [reply Engine/Annotations/Annotation.cs:1794]: Hmm, yes. I wanted to sneak in a nice simplification but I'll open a separate PR

**Review by rmadsen-ks — APPROVED** [review]
Looks good!

**Pull request merged**